pkg/api/types.go

```diff
@@ -1003,6 +1003,16 @@ type NodeSpec struct {
 
 	// Unschedulable controls node schedulability of new pods. By default node is schedulable.
 	Unschedulable bool `json:"unschedulable,omitempty"`
+
+	// Certificate is the public x.509 certificate tied to the node contained
+	// in a PEM-encoded byte array. It is presented by the node when it asks to
+	// join the cluster and can be refreshed if the certificate on the node is
+	// rotated. The node certificate is checked when initiating a connection to
+	// the node (to verify that the connection isn't being intercepted) and is
+	// presented by the node when it connects to the master (as a client
+	// certificate) to establish that the connection is coming from a trusted
+	// and identifiable source.
+	Certificate []byte `json:certificate,omitempty"`
 }
```

Comment:
What is the security value of having the master verifying that the node presented the exact same certificate, versus just verifying the cert is signed by a trusted CA?

I envision an admin being able to manually add nodes in the following way:
 - configure master with a pool trusted CA certs (but not necessarily the CA signing key: that might be kept somewhere more secure than the apiserver)
 - create nodes with keys signed by the trusted CA.  For physical nodes, this might be part of an initial install OS install process
 - manually create a nodeSpec which says that the node belongs to the cluster.
 - nodes, when they come up, contact the master and are immediately authenticated.
I realize that is not one of the two systems we designed, but it seems like a valid use case.